apps/arena/lib/arena/game_updater.ex

```diff
@@ -122,8 +89,69 @@ defmodule Arena.GameUpdater do
     state =
       state
       |> Map.put(:last_id, last_id)
-      |> Map.put(:entities, entities)
+      |> Map.put(:projectiles, projectiles)
 
     {:reply, :ok, state}
   end
+
+  ##################
+  # Private
+  ##################
+
+  # Game creation
+  defp new_game(game_id, players) do
+    new_game =
+      Physics.new_game(game_id)
+      |> Map.put(:last_id, 0)
+      |> Map.put(:players, %{})
+      |> Map.put(:projectiles, %{})
+
+    Enum.reduce(players, new_game, fn {_player_id, _client_id}, new_game ->
+      last_id = new_game.last_id + 1
+      players = new_game.players |> Map.put(last_id, Entities.new_player(last_id))
+
+      new_game
+      |> Map.put(:last_id, last_id)
+      |> Map.put(:players, players)
+    end)
+  end
+
+  # Move entities and add game fields
+  defp update_entities(entities) do
+    new_state = Physics.move_entities(entities)
+
+    Enum.reduce(new_state, %{}, fn {key, value}, acc ->
+      entity =
+        Map.get(entities, key)
+        |> Map.merge(value)
+        |> Map.put(:is_colliding, false)
```

Comment:
This is being set as false and saved in players' state, but when broadcasting to the client we're doing the collision calculation.
We should do the collision check before and saving the same value in both places (state and broadcast message).

Reply:
Fixed